lib/3scale/backend/transactor/status.rb

```diff
@@ -211,11 +194,35 @@ def with_report_and_hierarchy(reports)
           end
         end
 
-        def add_plan(xml, tag, plan_name)
+        def add_plan_section(xml, tag, plan_name)
           xml << "<#{tag}>"
           xml << plan_name.to_s.encode(xml: :text) << "</#{tag}>"
         end
 
+        def add_authorize_section(xml)
+          if authorized?
+            xml << '<authorized>true</authorized>'.freeze
+          else
+            xml << '<authorized>false</authorized><reason>'.freeze
+            xml << rejection_reason_text
+            xml << '</reason>'.freeze
+          end
+        end
+
+        def add_application_section(xml)
+          redirect_uri = application.redirect_url
+          xml << "<application><id>".freeze << "#{application.id}" << "</id>".freeze
+          xml << "<key>".freeze << "#{application.keys.first}" << "</key>".freeze
+          xml << "<#{@redirect_uri_field}>#{redirect_uri}</#{@redirect_uri_field}>"
```

Comment:
you can join here literals in a single string: `"</id><key>"`, `"</key><" as it will avoid a couple of string concatenation ops.